test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -643,6 +639,74 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
 		})
 	})
+
+	Context("Offlined CPU API", func() {
+		BeforeEach(func() {
+			By("Saving the old performance profile")
+			initialProfile = profile.DeepCopy()
+		})
+
+		It("[disruptive] should set offline cpus after deploy PAO", func() {
+			for _, node := range workerRTNodes {
+				onlineCPUCount, err := nodes.ExecCommandOnNode([]string{"nproc", "--all"}, &node)
+				Expect(err).ToNot(HaveOccurred())
+				onlineCPUInt, err := strconv.Atoi(onlineCPUCount)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(onlineCPUInt).Should(BeNumerically(">=", 3))
+			}
+
+			// Create new performance with offlined
+			reserved := performancev2.CPUSet("0")
+			isolated := performancev2.CPUSet("1")
+			offlined := performancev2.CPUSet("2-3")
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Enable UserLevelNetworking and add Devices in Profile")
+			profile.Spec.CPU = &performancev2.CPU{
+				Reserved: &reserved,
+				Isolated: &isolated,
+				Offlined: &offlined,
+			}
+			By("Updating the performance profile")
+			profiles.UpdateWithRetry(profile)
+
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+			By("Waiting for MCP being updated")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+			workerRTNodes = getUpdatedNodes()
+			//Check offlined cpus are setting correctly
+			for _, node := range workerRTNodes {
+				offlinedOutput, err := nodes.ExecCommandOnNode([]string{"cat", "/sys/devices/system/cpu/offline"}, &node)
+				Expect(err).ToNot(HaveOccurred())
+				offlinedCPUSet, err := cpuset.Parse(offlinedOutput)
+				offlinedCPUSetProfile, err := cpuset.Parse(string(offlined))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(offlinedCPUSet.Equals(offlinedCPUSetProfile))
+			}
+		})
+
+		AfterEach(func() {
+			By("Reverting the Profile")
+			spec, err := json.Marshal(initialProfile.Spec)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testclient.Client.Patch(context.TODO(), profile,
+				client.RawPatch(
+					types.JSONPatchType,
+					[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
+				),
+			)).ToNot(HaveOccurred())
```

Comment:
this should be enclosed in a `Eventually` or a retry loop of sorts. The revert must really be careful about not leaving the cluster in a corrupted state

Reply:
copied from here: https://github.com/openshift/cluster-node-tuning-operator/blob/0686c0a24caffb32567dbb516d0b847cc613b933/test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go#L628
should I also change it there?

Reply:
no, should be fixed on a separate PR

Reply:
I think we switched in the past to use patch instead of Update with retry on purpose : https://github.com/openshift-kni/performance-addon-operators/commit/a646abfa26253a6cef1c44285ae67e0f8d2a8c37

Reply:
Yes, nice catch. Then ok, if Patch avoids the conflict then no need to wrap in a loop.